spec/features/admin_applications/show_spec.rb

```diff
@@ -0,0 +1,112 @@
+          visit("/admin/applications/#{@application_1.id}")
+          
+          click_button("Approve #{@pet_1.name}")
+          
+          expect(current_path).to eq("/admin/applications/#{@application_1.id}")
+        end
+        
+        it "no longer has the approve button once that button has been clicked" do
+          visit("/admin/applications/#{@application_1.id}")
+          
+          click_button("Approve #{@pet_1.name}")
+          
+          expect(page).to_not have_content("Approve #{@pet_1.name}")
+        end
+
+        it "shows an indicator next to the pet that says that pet has been approved" do
+          visit("/admin/applications/#{@application_1.id}")
+
+          click_button("Approve #{@pet_1.name}")
+
+          expect(page).to have_content("#{@pet_1.name}: Approved")
+        end
+      end
+
+      describe "Rejecting a Pet for Adoption" do
+        it "shows a reject button next to every pet for that application" do
+          visit("/admin/applications/#{@application_1.id}")
+
+          expect(page).to have_button("Reject #{@pet_1.name}")
+        end
+        
+        it "directs back to the admin application show page when reject button is clicked" do
+          visit("/admin/applications/#{@application_1.id}")
+          
+          click_button("Reject #{@pet_1.name}")
+          
+          expect(current_path).to eq("/admin/applications/#{@application_1.id}")
+        end
+        
+        it "no longer has the reject button once that button has been clicked" do
+          visit("/admin/applications/#{@application_1.id}")
+          
+          click_button("Reject #{@pet_1.name}")
+          
+          expect(page).to_not have_content("Reject #{@pet_1.name}")
+        end
+
+        it "shows an indicator next to the pet that says that pet has been rejected" do
+          visit("/admin/applications/#{@application_1.id}")
+
+          click_button("Reject #{@pet_1.name}")
+
+          expect(page).to have_content("#{@pet_1.name}: Rejected")
+        end
+      end
+
+      describe "Approving/Rejecting doesn't affect pet's on other show pages" do
+        it "when I approve a pet on one show page it doesn't affect other show page pets" do
+          visit("/admin/applications/#{@application_1.id}")
+
+          click_button("Approve #{@pet_1.name}")
+
+          visit("/admin/applications/#{@application_2.id}")
+
+          expect(page).to have_button("Approve #{@pet_1.name}")
+          expect(page).to have_button("Reject #{@pet_1.name}")
+        end
+
+        it "when I reject a pet on one show page it doesn't affect other show page pets" do
+          visit("/admin/applications/#{@application_1.id}")
+
+          click_button("Reject #{@pet_1.name}")
+
+          visit("/admin/applications/#{@application_2.id}")
+
+          expect(page).to have_button("Approve #{@pet_1.name}")
+          expect(page).to have_button("Reject #{@pet_1.name}")
+        end
+      end
+    end
+  end
+end
```

Comment:
Overall, I couldn't find anything in this test suite that I could call out specifically. I always like when assertions use anchor text to reduce ambiguity on the page and prevent false positives.